gemma-core/src/main/java/ubic/gemma/core/ontology/providers/GeneOntologyServiceImpl.java

```diff
@@ -816,7 +820,7 @@ private synchronized void initializeGeneOntology() {
         if ( GeneOntologyServiceImpl.running.get() )
             return;
 
-        Thread loadThread = new Thread( new Runnable() {
+        this.taskExecutor.execute( new Runnable() {
             @Override
             public void run() {
                 GeneOntologyServiceImpl.running.set( true );
```

Comment:
This should use a `Future` instead... We need to get rid of those `AtomicBoolean` & `Thread.sleep` loop patterns.